tratto/src/main/java/star/tratto/util/javaparser/JavaParserUtils.java

```diff
@@ -49,6 +74,63 @@ public static JavaParser getJavaParser() {
         return javaParser;
     }
 
+    /**
+     * Gets the type of the given java expression. For example, given the
+     * expression "jpClass.getMethods().get(0)", this method outputs,
+     * "ResolvedMethodDeclaration".
+     *
+     * @param jpClass the declaring class
+     * @param jpCallable the method in which the type is used
+     * @param methodArgs the arguments of the method
+     * @param expression a Java expression (e.g. "jpClass.getMethods()")
+     * @return the resolved type of the expression
+     * @throws ResolvedTypeNotFound if the type is not found
+     */
+    public static ResolvedType getResolvedTypeOfExpression(
+            TypeDeclaration<?> jpClass,
+            CallableDeclaration<?> jpCallable,
+            List<Triplet<String, String, String>> methodArgs,
```

Comment:
As I mentioned in the issue, I'm not very familiar with records. Based on this comment, if we were to replace tuple classes with records, would we end up with a dozen or more new classes? We make heavy usage of tuples, including Quintets and Sextets. How would you call a Sextet in a context such as the following?:

```java
     * @param jpTag a sextet of tag information, including: file source code,
     *              JavaParser class, JavaParser method/constructor, oracle
     *              type, name, and content.
     * @return the original tag in source code as a String.
     */
    public static String reconstructTag(
            Sextet<String, TypeDeclaration<?>, CallableDeclaration<?>, OracleType, String, String> jpTag
```

With the Sextet, at least, you know the types of the elements stored in the structure (String, TypeDeclaration, etc.).

Reply:
Yes, we would end up with a dozen or more record classes.  This is a good thing!  Each record declaration would be very short, so there would be little code clutter.

Each record would have a (short!) name that indicates its meaning, which is not available when using a Sextet.

Another advantage is that getters have informative names.  `mySextet.getValue3()` is inscrutible and misleading.  When not indexing into a list or array, one starts counting at one rather than zero.  However, `getValue3()` is the *fourth* element of the sextet!

To create a record instance, you call `new MyRecordName(arg1, arg2, ...)`.  So the construction is no different than for a sextet -- just clearer because of the use of a meaningful name.

Reply:
Sounds good, thanks a lot for the advice. This change will affect many parts of the code. To make sure that no regressions are introduced in the process (and with any other of the latest change requests), please check my last comment in this thread, @mernst, @ezackr.